app/controllers/questions_controller.rb

```diff
@@ -28,10 +36,24 @@ def update
   end
 
   def destroy
+    if @question.user_id != @user_id
+      return render json: { message: 'Not authorized to delete the question' },
+                    status: :unauthorized
+    end
+
     @question.destroy
     render status: :no_content
   end
 
+  def set_user_id
+    @user_id = @current_user.id
+  end
```

Comment:
too much abstraction to access a single variable. Just use @current_user.id directly whenever needed.